### ISSUE TITLE
required by a bound in `component_view`

### ISSUE THREAD

**Issue by qtisan** [eboody/styled#1]
can not pass in compile.

~~//

```
the trait bound `&fn(leptos_reactive::scope::Scope, StyleProps) -> impl leptos_dom::IntoView {Style}: ComponentConstructor<_>` is not satisfied
the trait `ComponentConstructor<_>` is not implemented for `&fn(leptos_reactive::scope::Scope, StyleProps) -> impl leptos_dom::IntoView {Style}`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Flennon%2FDocuments%2FGitHub%2Fmoerae_view%2Fsrc%2Fapplication.rs)
lib.rs(21, 9): Actual error occurred here
lib.rs(21, 9): Error originated from macro call here
lib.rs(296, 13): required by a bound in `component_view`

the trait bound `fn(leptos_reactive::scope::Scope, StyleProps) -> impl leptos_dom::IntoView {Style}: leptos::Component<_>` is not satisfied
the trait `leptos::Component<_>` is not implemented for fn item `fn(leptos_reactive::scope::Scope, StyleProps) -> impl leptos_dom::IntoView {Style}`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B1%5D?1#file%3A%2F%2F%2FUsers%2Flennon%2FDocuments%2FGitHub%2Fmoerae_view%2Fsrc%2Fapplication.rs)
lib.rs(21, 9): Actual error occurred here
lib.rs(21, 9): Error originated from macro call here
lib.rs(289, 15): required by a bound in `component_props_builder`
```

<img width="572" alt="截屏2023-07-25 上午12 40 27" src="https://github.com/eboody/styled/assets/8434908/0a8b5ae0-0137-4a38-a5e2-e71b9beb4077">

**Comment by qtisan** [comment]
```toml
[dependencies]
leptos = { version = "0.4.5", features = ["csr", "nightly"] }
styled = "0.1.36"
stylist = "0.12.1"
```

**Comment by pitasi** [comment]
Yes, I have the same issue. This crate was most likely broken by newer versions of leptos.

**Comment by cnd4** [comment]
I have the same issue here too.

```toml
leptos = { version = "0.4", features = ["nightly"] }
styled = "0.1.36"
stylist = "0.12.1"
```

Note that I use `cargo leptos watch`, I have SSR and hydratation enabled.

**Comment by eboody** [comment]
Thanks for this! Just got back from vacation yesterday so apologies for the delay but I'll check this out right now.

**Comment by eboody** [comment]
Thanks for letting me know guys! This should be addressed in version `1.37`

**Comment by eboody** [comment]
Thanks for letting me know guys! This should be addressed in version 1.37. Do let me know how it goes!
